Scripts/Editor/UI/EditorInputDialog.cs

```diff
@@ -0,0 +1,178 @@
+using System;
+using UnityEditor;
+using UnityEngine;
+
+namespace Anvil.Unity.Editor.UI
+{
+    /// <summary>
+    /// Helper class to show an Input dialog in the editor.
+    /// </summary>
+    public class EditorInputDialog : EditorWindow
+    {
+        private const string CONTROL_NAME_INPUT_TEXT = "EditorInputDialog_INPUT_TEXT";
+
+        /// <summary>
+        /// Shows the Input Dialog
+        /// </summary>
+        /// <param name="title">The title of the Dialog window</param>
+        /// <param name="description">The description text to communicate what the user should do</param>
+        /// <param name="inputTextLabel">A label to describe the input text</param>
+        /// <param name="okBtnText">What the "OK" button should read</param>
+        /// <param name="cancelBtnText">What the "Cancel" button should read</param>
+        /// <param name="defaultInputText">The default text in the input box if needed</param>
+        /// <returns>
+        /// The input text the user entered.
+        /// If the user pressed the "OK" button, this will be whatever they have entered.
+        /// If the user pressed the "Cancel" button, this will be string.Empty.
+        /// </returns>
+        public static string Show(
+            string title,
+            string description,
+            string inputTextLabel,
+            string okBtnText,
+            string cancelBtnText,
+            string defaultInputText = null)
+        {
+            Vector2 maxPos = GUIUtility.GUIToScreenPoint(new Vector2(Screen.width, Screen.height));
+
+            string result = string.Empty;
+
+            EditorInputDialog window = CreateInstance<EditorInputDialog>();
+            window.Init(
+                maxPos,
+                title,
+                description,
+                inputTextLabel,
+                okBtnText,
+                cancelBtnText,
+                defaultInputText);
+            window.OnComplete += (resultString) => result = resultString;
```

Comment:
Since this method doesn't seem to return until the window closes should we just read the input state off the dialog? No need to have an event.